samples/ck-rx65n/basic-sample/src/iotconnect_app.c

```diff
@@ -0,0 +1,267 @@
+    printf("Sent CMD ack: %s\r\n", ack);
+    iotconnect_sdk_send_packet(ack);
+    free((void*) ack);
+}
+
+static void on_command(IotclEventData data) {
+	char* context = NULL;
+    char *command = iotcl_clone_command(data);
+    if (NULL != command) {
+    	char* token = strtok_r(command, " ", &context);
+    	if(!strncmp(token, "led", 3)) {
+    		token = strtok_r(NULL, " ", &context);
+    		RX65N_LED_STATE state = (token[0] == '1') ||
+    								(strncmp(token, "true", 4) == 0 ) ?
+    								ON : OFF;
+    		set_led(LED2, state);
+    		set_led(LED1, state);
+    		command_status(data, true, command, "LED set");
+    	}
+    	else {
+    		command_status(data, false, command, "Not implemented");
+    	}
+        free((void*) command);
+    } else {
+        command_status(data, false, "?", "Internal error");
+    }
+}
+
+static void on_connection_status(IotConnectConnectionStatus status) {
+    // Add your own status handling
+    switch (status) {
+    case MQTT_CONNECTED:
+        printf("IoTConnect Client Connected\r\n");
+        break;
+    case MQTT_DISCONNECTED:
+        printf("IoTConnect Client Disconnected\r\n");
+        break;
+    default:
+        printf("IoTConnect Client ERROR\r\n");
+        break;
+    }
+    if (NULL != auth_driver_context) {
+    	release_sw_der_auth_driver(auth_driver_context);
+    }
+}
+
+static void publish_telemetry() {
+    const char *str;
+    IotclMessageHandle msg = iotcl_telemetry_create(iotconnect_sdk_get_lib_config());
+
+    // Optional. The first time you create a data point, the current timestamp will be automatically added
+    // TelemetryAddWith* calls are only required if sending multiple data points in one packet.
+    iotcl_telemetry_add_with_iso_time(msg, iotcl_iso_timestamp_now());
+    iotcl_telemetry_set_string(msg, "version", APP_VERSION);
+//   iotcl_telemetry_set_number(msg, "cpu", 3.123); // test floating point numbers
+    // random number 0-100, cast to int so that it removes decimals in json
+    iotcl_telemetry_set_number(msg, "random", (int)((double)rand() / (double)RAND_MAX * 100.0));
+
+    iotcl_telemetry_set_bool(msg, "button", read_user_switch());
+
+//    sensors_add_telemetry(msg);
+    str = iotcl_create_serialized_string(msg, false);
+    iotcl_telemetry_destroy(msg);
+    printf("Sending: %s\r\n", str);
+    if (NULL != str) {
+        iotconnect_sdk_send_packet(str); // underlying code will report an error
+        iotcl_destroy_serialized(str);
+    }
+}
+
+/* Include the sample.  */
+bool app_startup(NX_IP *ip_ptr, NX_PACKET_POOL *pool_ptr, NX_DNS *dns_ptr) {
+    printf("Starting App Version %s\r\n", APP_VERSION);
+    IotConnectClientConfig *config = iotconnect_sdk_init_and_get_config();
+    azrtos_config.ip_ptr = ip_ptr;
+	azrtos_config.pool_ptr = pool_ptr;
+	azrtos_config.dns_ptr = dns_ptr;
+
+    config->cpid = IOTCONNECT_CPID;
+    config->env = IOTCONNECT_ENV;
+#ifdef IOTCONNECT_DUID

```

Comment:
Unfair to raise it on this specific branch, but...

If 
`#define IOTCONNECT_DUID "" `
don't think this `#ifdef` test works?

Probably need a run-time test (C preprocessor not easy to find empty strings?).

Reply:
@nmatthews-witekio I think that that code means to say if you did not define DUID, you want to use dynamically generated DUID (not all projects actually do this). If you define a blank one, it will abviously not work. We don't need to check for blank.